engines/dfc_provider/app/services/quantitative_value_builder.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Describes the quantity contained in a product, for example:
+#
+# - 1 piece of apple, usually meaning the whole fruit
+# - 2 litres of milk, for example in a big bottle or pouch
+# - 750 grams of bread, for example a loaf
+#
+# The DFC also supports specific units like loafs and cans but we don't have
+# standardised data within OFN to map to these types.
+class QuantitativeValueBuilder < DfcBuilder
+  def self.quantity(variant)
+    DataFoodConsortium::Connector::QuantitativeValue.new(
+      unit: unit(variant),
+      value: variant.unit_value,
+    )
+  end
+
+  def self.unit(variant)
+    case variant.product.variant_unit
+    when "volume"
+      DfcLoader.connector.MEASURES.UNIT.QUANTITYUNIT.LITRE
+    when "weight"
+      DfcLoader.connector.MEASURES.UNIT.QUANTITYUNIT.GRAM
```

Comment:
Uh, I should double-check that. I thought that we always store grams in the database but display g, kg or tonne depending on the scale. I'll have a look.

Reply:
I finally found a spec that shows that I was right:
https://github.com/openfoodfoundation/openfoodnetwork/blob/93c0a064032e414ed947ce96fd974535272a17f3/spec/services/variant_units/option_value_namer_spec.rb#L60-L86

If you have a unit_value of 1 and a variant_unit_scale of 1000 then it would still be 1g, displayed as 0.001kg. The example above actually has a value of 1000 with a scale of 1000 which then becomes 1kg, not 1000kg. The scale is not a multiplier, it indicates the scale of the unit to display. It's used to divide by when converting to the displayed unit.

Do you agree seeing these specs?

Reply:
Ah, so sorry to send you on a wild goose chase! Yes, this makes sense and on re-reading the importer, I can see that is consistent.
So the variant_unit_scale is purely a formatting setting. Seems like a strange way to do it to me, but moving on...